api/instance/v1/security_group_utils_test.go

```diff
@@ -0,0 +1,57 @@
+package instance
+
+import (
+	"testing"
+
+	"github.com/scaleway/scaleway-sdk-go/internal/testhelpers"
+	"github.com/scaleway/scaleway-sdk-go/internal/testhelpers/httprecorder"
+	"github.com/scaleway/scaleway-sdk-go/utils"
+)
+
+func TestAPI_UpdateSecurityGroup(t *testing.T) {
+
+	client, r, err := httprecorder.CreateRecordedScwClient("security-group-test")
+	testhelpers.Ok(t, err)
+	defer func() {
+		testhelpers.Ok(t, r.Stop()) // Make sure recorder is stopped once done with it
+	}()
+
+	instanceAPI := NewAPI(client)
+
+	var (
+		organization = "b3ba839a-dcf2-4b0a-ac81-fc32370052a0"
+	)
+
+	createResponse, err := instanceAPI.CreateSecurityGroup(&CreateSecurityGroupRequest{
+		Zone:                  utils.ZoneFrPar1,
+		Name:                  "name",
+		Description:           "description",
+		Stateful:              true,
+		InboundDefaultPolicy:  SecurityGroupPolicyAccept,
+		OutboundDefaultPolicy: SecurityGroupPolicyDrop,
+		Organization:          organization,
+	})
+
+	testhelpers.Ok(t, err)
+
+	accept := SecurityGroupPolicyAccept
+	drop := SecurityGroupPolicyDrop
+	f := false
+
+	updateResponse, err := instanceAPI.UpdateSecurityGroup(&UpdateSecurityGroupRequest{
+		Zone:                  utils.ZoneFrPar1,
+		SecurityGroupID:       createResponse.SecurityGroup.ID,
+		Name:                  utils.String("new_name"),
+		Description:           utils.String("new_description"),
+		Stateful:              &f,
+		InboundDefaultPolicy:  &drop,
+		OutboundDefaultPolicy: &accept,
+	})
+
+	testhelpers.Ok(t, err)
+	testhelpers.Equals(t, "new_name", updateResponse.SecurityGroup.Name)
+	testhelpers.Equals(t, "new_description", updateResponse.SecurityGroup.Description)
+	testhelpers.Equals(t, SecurityGroupPolicyDrop, updateResponse.SecurityGroup.InboundDefaultPolicy)
+	testhelpers.Equals(t, SecurityGroupPolicyAccept, updateResponse.SecurityGroup.OutboundDefaultPolicy)
+	testhelpers.Equals(t, false, updateResponse.SecurityGroup.Stateful)
+}
```

Comment:
You should also delete the resource you create. Otherwise we'll have dangling resources when updating the cassettes.